### ISSUE TITLE
Spawned Process doesn't pass node arguments down

### ISSUE THREAD

**Issue by wesbos** [gatsbyjs/gatsby#24925]
I'm using `esm` to use ES modules in gatsby-[ssr,node,config].js and I have used it like so for over a year:

```
"build": "npx --node-arg '-r esm' gatsby build",
"develop": "npx --node-arg '-r esm' gatsby develop",
```

Something changed recently and none of these projects will build, I get the errors about not being allowed to use import/export.

A replication can be found here: https://github.com/reaktivo/gatsby-esm/issues/1 

@marcysutton confirmed the issue: https://twitter.com/marcysutton/status/1270828197588787200

@ascorbic has suggested to try NODE_OPTIONS env variable because apparently develop now runs in a spawned processs? https://twitter.com/ascorbic/status/1270836362602741760

@mxstbr has said it was him that broke it 🤣 https://twitter.com/mxstbr/status/1270836603410419716

Thanks a ton! Let me know if I can offer any more help.

**Comment by mxstbr** [comment]
This very likely broke because of #22759, similar to #24349. 

I'm wondering if there's a way to pass all node options through to the child process in order to preserve all this functionality...

**Comment by wesbos** [comment]
thanks Max - I've tried getting it working with NODE_OPTIONS but that doesn't seem to work. Does anyone know if I'm doing this right? Or maybe it just doesn't get passed down at all?

```
"develop": "NODE_OPTIONS='-r esm' gatsby develop",
```

**Comment by wesbos** [comment]
The above worked with the latest version of gatsby! Thanks

**Comment by bdenham** [comment]
I found this issue to be the only one related to ESM support that was still open and recently added to triage. So I’ll ask my question here. 

Does version 3.6.x no longer support the importing of pure ESM modules into `gatsby-configure.js` using the `esm` modules method that Wes described here and in the other Closed ESM issues?

I’ve tried every method I could find from various sites, blogs, forums to find a technique that would work, using the `esm` module, using the dynamic import within an async IIFE, using Node’s now full support of ESM modules, and so on. All failed to move me forward with my use case: I want to use the `unist-util-visit` module in my `gatsby-config.js` transformer to walk the mdxAST and create several content objects (records) based on certain node types from each mdx file, which I then send my Algolia search index. Without the unist modules, it’s not so easy.

I did just now see someone else confirm in another issue here that the 2.x versions of the `unist-util-visit` module did work for him, but I haven’t tried it yet. I hope it works for me as well. Regardless, I think a lot of people would like to see Gatsby make this possible again. Thanks!

Thanks!
